src/WL/Client.hs

```diff
@@ -0,0 +1,4 @@
+{-# LANGUAGE EmptyDataDecls #-}
+module WL.Client where
+
+data WL_client
```

Comment:
Shouldn’t this type (and `WL_global`) should be associated with their corresponding C type? Either using a `CTYPE` pragma or by using `{{ struct … }}`.

Reply:
I couldn't find the definitions so I thought they were supposed to be opaque

Reply:
Ah, looks like you’re right, although I’m pretty sure it’s possible to add `CTYPE` anyway.

Reply:
If there's no Header I'm not sure what <X> and <Y> should be in
```
data {-# CTYPE "<X>" "<Y"> #-} wL_client
```

Reply:
…yeah, actually, that makes sense. I got confused.